docs/contributing/3.0-development/app-layout.rst

```diff
@@ -0,0 +1,241 @@
+    │   │   │   ├── __init__.py
+    │   │   │   └── ...
+    │   │   └── __init__.py
+    │   ├── manage.py
+    │   ├── migrations
+    │   ├── models
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── pagination.py
+    │   ├── response.py
+    │   ├── serializers
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── settings.py
+    │   ├── tasks
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   ├── templates
+    │   │   └── ...
+    │   ├── tests
+    │   │   └── ...
+    │   ├── urls.py
+    │   ├── viewsets
+    │   │   ├── __init__.py
+    │   │   └── ...
+    │   └── wsgi.py
+    └── __init__.py
+
+
+The contents of this package are documented in detail in the :doc:`../platform_api/index`
+documentation. Details how this package is organized can be found
+below, along with information about some of the modules found in this namespace.
+
+.. tip::
+
+    This tree is incomplete, and maybe be out of date. Only the most notable and
+    durable modules have been listed; the most complete listing of modules in this
+    namespace will always be the :doc:`../platform_api/index` documentation.
+
+
+Module Imports
+--------------
+
+For modules in the ``pulp.app`` namespace that are large and capture behaviors
+across multiple concerns of pulp, such as our models, we have broken those
+package up into multiple subpackages. All public identifiers and objects defined
+in submodules are then collected into that module's ``__init__.py``, from which
+they will be imported by other Pulp Platform modules.
+
+Using :mod:`pulp.app.models` as an example, this means that when breaking up the
+``models`` module in ``pulp.app``, the following things are true:
+
+* No models are defined in the ``__init__.py`` of ``pulp.app.models``.
+* All models are defined in submodules located in the ``pulp.app.models`` module
+  directory (where its ``__init__.py`` can be found).
+* The `__init__.py`` in ``pulp.app.models`` should consist only of import statements,
+  ordered to prevent any circular import issues that may result based on the imports
+  that are done in any included submodules.
+* Any models defined in submodules in ``pulp.app.models`` namespace must be imported
+  from the ``pulp.app.models`` namespace, not the submodule in which they are defined.
+  Yes: ``from pulp.app.models import PulpModel``,
+  No: ``from pulp.app.models.pulp import PulpModel``.
+* When adding new models, they must be imported into the ``pulp.app.models``
+  ``__init__.py``, so that they are available to be imported by any other Pulp Platform
+  components that use them from the ``pulp.app.models`` namespace.
+* Imports done inside any submodules should be relative, e.g.
+  ``from .submodule import identifier`` or ``from . import submodule``, avoiding the
+  creation of circular imports.
+* Imports done inside the module's ``__init__.py`` should be relative and explict, e.g.
+  ``from .submodule import identifier1, identifier2``, NOT
```

Comment:
Could you put these examples on new lines and include Yes: and No: at the beginning like line 99? Negative examples should be impossible to misread as positive examples.